src/pepr/operator/controllers/keycloak/authservice/authservice.ts

```
@@ -27,41 +31,82 @@ export const log = setupLogger(Component.OPERATOR_AUTHSERVICE);
+    const waypointId = sso.clientId;
+    // Get the full waypoint name for authorization policies
+    const fullWaypointName = getWaypointName(waypointId);
+
     await reconcileAuthservice(
       { name: sso.clientId, action: Action.AddClient, client },
       sso.enableAuthserviceSelector!,
       pkg,
+      isAmbient,
+      fullWaypointName,
     );
   }
 
   const authserviceClients = authServiceClients.map(client => client.clientId);
 
-  await purgeAuthserviceClients(pkg, authserviceClients);
-
+  await purgeAuthserviceClients(pkg, authserviceClients, isAmbient);
+  // Clean up any existing waypoint resources if SSO is not configured
+  await purgeOrphans(
+    (pkg.metadata?.generation ?? 0).toString(),
+    pkg.metadata.namespace,
+    pkg.metadata.name,
+    K8sGateway,
+    log,
+  );
   return authserviceClients;
 }
 
 export async function purgeAuthserviceClients(
   pkg: UDSPackage,
   newAuthserviceClients: string[] = [],
+  isAmbient = false,
 ) {
-  // compute set difference of pkg.status.authserviceClients and authserviceClients using Ramda
   R.difference(pkg.status?.authserviceClients || [], newAuthserviceClients).forEach(
     async clientId => {
       log.info(`Removing stale authservice chain for client ${clientId}`);
-      await reconcileAuthservice({ name: clientId, action: Action.RemoveClient }, {}, pkg);
+      // Use client ID directly as the waypoint ID
+      const waypointId = clientId;
+      // Get the full waypoint name for authorization policies
+      const fullWaypointName = getWaypointName(waypointId);
+      await reconcileAuthservice(
+        { name: clientId, action: Action.RemoveClient },
+        {},
+        pkg,
+        isAmbient,
+        fullWaypointName,
+      );
+
+      // Clean up waypoint labels
+      if (isAmbient && pkg.metadata?.namespace) {
+        await cleanupWaypointLabels(pkg.metadata.namespace, fullWaypointName);
+      }
```

There's a small edge case here. If you switch a package from ambient -> sidecar, we'd want to make sure we cleanup waypoint labels. In practice this cleanup isn't hit though because it's conditional on the _current_ mode being ambient. We might be able to safely remove the `isAmbient` check since we'd want to cleanup these labels if they are in sidecar mode (and the labels exist)?

---

I also noticed this doesn't properly purge waypoint labels on deletion of the package CR. In the finalizer this purge function is called without the `isAmbient` parameter, meaning it's always false and this is skipped - https://github.com/defenseunicorns/uds-core/blob/implement-aaa/src/pepr/operator/reconcilers/package-reconciler.ts#L196-L199

Defaulting `isAmbient` to `false` might be hiding some places like this where we aren't passing in the mesh mode properly.

---

I did just run through this with another scenario and there may be another edge case if you keep the client around but change the selector - not sure how to handle that one though 😬. Currently this logic all goes off of client removal so it's a bit harder to tell when the client stays around but the selector changes.